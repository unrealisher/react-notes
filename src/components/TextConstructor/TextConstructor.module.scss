.wrapper {
  width: 100%;
  display: flex;
  flex-direction: column;
}

.textarea {
  box-sizing: border-box;
  resize: none;
  flex-grow: 1;
  padding: 12px;
  border-radius: 15px;
}
