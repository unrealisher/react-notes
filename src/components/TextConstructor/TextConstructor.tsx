import React, { useRef } from "react";
import cn from "classnames";

import styles from "./TextConstructor.module.scss";

interface IProps {
  wrapper?: string;
  setText: Function;
}

const TextConstructor = (props: IProps): JSX.Element => {
  const { wrapper, setText } = props;
  const textAreaRef = useRef<HTMLTextAreaElement>(null);
  return (
    <div className={cn(styles.wrapper, wrapper ? wrapper : undefined)}>
      <label htmlFor="textarea">Текст заметки:</label>
      <textarea
        id="textarea"
        className={styles.textarea}
        ref={textAreaRef}
        onChange={() => {
          if (textAreaRef.current) setText(textAreaRef.current.value);
        }}
      />
    </div>
  );
};

export default TextConstructor;
