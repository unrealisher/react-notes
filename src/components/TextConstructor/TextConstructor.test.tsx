import React from "react";
import { shallow } from "enzyme";
import { TextConstructor } from "./TextConstructor";

describe("TextConstructor", () => {
  const text = "Hello World!";
  const setText = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <TextConstructor text={text} setText={setText} />
    );
    expect(component).toMatchSnapshot();
  });
});
