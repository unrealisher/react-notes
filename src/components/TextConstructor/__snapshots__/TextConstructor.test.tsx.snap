// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextConstructor correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextConstructor
    setText={[Function]}
    text="Hello World!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="textarea"
        >
          Текст заметки:
        </label>,
        <textarea
          className="textarea"
          id="textarea"
          name="textarea"
          onChange={[Function]}
          value="Hello World!"
        />,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Текст заметки:",
          "htmlFor": "textarea",
        },
        "ref": null,
        "rendered": "Текст заметки:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "textarea",
          "id": "textarea",
          "name": "textarea",
          "onChange": [Function],
          "value": "Hello World!",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "textarea",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="textarea"
          >
            Текст заметки:
          </label>,
          <textarea
            className="textarea"
            id="textarea"
            name="textarea"
            onChange={[Function]}
            value="Hello World!"
          />,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Текст заметки:",
            "htmlFor": "textarea",
          },
          "ref": null,
          "rendered": "Текст заметки:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "textarea",
            "id": "textarea",
            "name": "textarea",
            "onChange": [Function],
            "value": "Hello World!",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "textarea",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
